trace-utils/src/send_data/mod.rs

```diff
@@ -337,53 +342,72 @@ impl SendData {
 
     async fn send_with_protobuf(&self) -> SendDataResult {
         let mut result = SendDataResult::default();
-        let mut chunks: u64 = 0;
-        for tracer_payload in &self.tracer_payloads {
-            chunks += u64::try_from(tracer_payload.chunks.len()).unwrap();
-        }
-        let agent_payload = construct_agent_payload(self.tracer_payloads.clone());
-        let serialized_trace_payload = match serialize_proto_payload(&agent_payload)
-            .context("Failed to serialize trace agent payload, dropping traces")
-        {
-            Ok(p) => p,
-            Err(e) => return result.error(e),
-        };
-
-        result
-            .update(
-                self.send_payload(
-                    HEADER_CTYPE_PROTOBUF,
-                    serialized_trace_payload,
-                    chunks,
-                    None,
-                )
-                .await,
-            )
-            .await;
+        let chunks = u64::try_from(self.tracer_payloads.size()).unwrap();
+
+        match &self.tracer_payloads {
+            TracerPayloadCollection::V07(payloads) => {
+                let agent_payload = construct_agent_payload(payloads.to_vec());
+                let serialized_trace_payload = match serialize_proto_payload(&agent_payload)
+                    .context("Failed to serialize trace agent payload, dropping traces")
+                {
+                    Ok(p) => p,
+                    Err(e) => return result.error(e),
+                };
+
+                result
+                    .update(
+                        self.send_payload(
+                            HEADER_CTYPE_PROTOBUF,
+                            serialized_trace_payload,
+                            chunks,
+                            None,
+                        )
+                        .await,
+                    )
+                    .await;
 
-        result
+                result
+            }
+            _ => result,
+        }
     }
 
     async fn send_with_msgpack(&self) -> SendDataResult {
         let mut result = SendDataResult::default();
-
         let mut futures = FuturesUnordered::new();
-        for tracer_payload in self.tracer_payloads.iter() {
-            let chunks = u64::try_from(tracer_payload.chunks.len()).unwrap();
-            let additional_payload_headers =
-                Some(HashMap::from([(HEADER_DD_TRACE_COUNT, chunks.to_string())]));
-
-            let payload = match rmp_serde::to_vec_named(&tracer_payload) {
-                Ok(p) => p,
-                Err(e) => return result.error(anyhow!(e)),
-            };
-            futures.push(self.send_payload(
-                HEADER_CTYPE_MSGPACK,
-                payload,
-                chunks,
-                additional_payload_headers,
-            ));
+
+        match &self.tracer_payloads {
```

Comment:
Here I've kept the compatibility to send v07 payloads serialized as a msgpack. Is something we want to do? sending v07 to the agent? WDYT?

Reply:
> Is something we want to do?

We don't plan ever doing this honestly